cmd/db/db.go

```diff
@@ -0,0 +1,240 @@
+package db
```

Comment:
I’m not sure about this `cmd/db`. For me `cmd` should be consumers of the API not providers and I find depending on a cmd package from another cmd package a bit counterintuitive.

what do you think of something like this?
```
pkg/mysql
├── client
│   ├── direct_client.go
│   └── exec_client.go
├── config.go
├── mysql.go
└── topology
    └── topology.go
```

and consumers can create instances like:
```
client.NewExecClient(...)
client.NewDirectClient(...)
```

Reply:
```
cmd/mysql/client.go `ReplicationManager`
- StartReplication(ctx context.Context, host, replicaPass string, port int32) error
- StopReplication(ctx context.Context) error
- ResetReplication(ctx context.Context) error
- ReplicationStatus(ctx context.Context) (db.ReplicationStatus, string, error)                  
- IsReplica(ctx context.Context) (bool, error)
- DisableSuperReadonly(ctx context.Context) error
- IsReadonly(ctx context.Context) (bool, error)
- ReportHost(ctx context.Context) (string, error)
- CloneInProgress(ctx context.Context) (bool, error)
- Clone(ctx context.Context, donor, user, pass string, port int32) error
- DumbQuery(ctx context.Context)
- GetMemberState(ctx context.Context, host string) (db.MemberState, error)                       
- CheckIfInPrimaryPartition(ctx context.Context) (bool, error)
- EnableSuperReadonly(ctx context.Context) error
- Close() error


pkg/mysql/client/replication.go `ReplicationManager`
- ChangeReplicationSource(ctx context.Context, host, replicaPass string, port int32) error
- ReplicationStatus(ctx context.Context) (ReplicationStatus, string, error)                      
- GetGroupReplicationPrimary(ctx context.Context) (string, error)
- GetGroupReplicationReplicas(ctx context.Context) ([]string, error)
- GetMemberState(ctx context.Context, host string) (MemberState, error)                          


pkg/mysql/client/users.go `UserManager`
- DiscardOldPasswords(ctx context.Context, users []mysql.User) error
- UpdateUserPasswords(ctx context.Context, users []mysql.User) error
```

As you can see the structure of these three components, there are three different and distinct things, meaning, there is nothing in common between them so we could have a single interface with different implementations. If we want some interfaces they would be like three different ones with implementation for each of them and that is not necessary. If we want something that you suggested, we would more-or-less have something similar that we had before, and I wouldn't like that.

Thats why I still suggest a structure like this, with only types that expose certain methods for these three use cases. The thing that I think is good from the suggestion is the location of everything. I like more to have `pkg/mysql/client/...` rather than `pkg/db`, much better.

And regarding your concern Ege about like a cmd importing `cmd/db`, that is OK and it should not necesseraliy come from `pkg/...`, because it is related only to `cmd`. Imagine having some common util that is needed for different stuff in `cmd`, you wouldn't put it in `pkg`. And because I also wanted to confirm this opinion of mine, I looked at Docker and Kubernetes source code and this is a common thing there as well.